shared/src/models/utils/spatial-tree.ts

```diff
@@ -0,0 +1,176 @@
+     */
+    @IsObject()
+    public readonly spatialTreeAsJSON: ImmutableJsonObject = new PointRBush(
+        SpatialTree.rBushNodeSize
+    ).toJSON();
+
+    static readonly create = getCreate(this);
+
+    /**
+     * @deprecated Use {@link create} instead
+     */
+    // eslint-disable-next-line @typescript-eslint/no-useless-constructor, @typescript-eslint/no-empty-function
+    constructor() {}
+
+    /**
+     * @param spatialTree inlcuding a {@link PointRBush} saved in {@link spatialTreeAsJSON} as an {@link ImmutableJsonObject}
+     * @returns a new {@link PointRBush} with all the methods to search, add etc. elements in it
+     */
+    private static getPointRBush(spatialTree: SpatialTree) {
+        // PointRBush.fromJSON() runs in O(1)
+        return new PointRBush(this.rBushNodeSize).fromJSON(
+            spatialTree.spatialTreeAsJSON
+        );
+    }
+
+    /**
+     * Writes the {@link PointRBush} as an {@link ImmutableJsonObject} into {@link spatialTree}
+     */
+    private static savePointRBush(
+        spatialTree: Mutable<SpatialTree>,
+        pointRBush: PointRBush
+    ) {
+        // PointRBush.toJSON() runs in O(1)
+        spatialTree.spatialTreeAsJSON = pointRBush.toJSON();
+    }
+
+    public static addElement(
+        spatialTree: Mutable<SpatialTree>,
+        elementId: UUID,
+        position: Position
+    ) {
+        const pointRBush = this.getPointRBush(spatialTree);
+        pointRBush.insert({
+            position,
+            id: elementId,
+        });
+        this.savePointRBush(spatialTree, pointRBush);
+    }
+
+    public static removeElement(
+        spatialTree: Mutable<SpatialTree>,
+        elementId: UUID,
+        position: Mutable<Position> | Position
+    ) {
+        const pointRBush = this.getPointRBush(spatialTree);
+        pointRBush.remove(
+            {
+                position,
+                id: elementId,
+            },
+            (a, b) => a.id === b.id
+        );
+        this.savePointRBush(spatialTree, pointRBush);
+    }
+
+    public static moveElement(
+        spatialTree: Mutable<SpatialTree>,
+        elementId: UUID,
+        startPosition: Mutable<Position> | Position,
+        targetPosition: Position
+    ) {
+        // TODO: use the move function from RBush, when available: https://github.com/mourner/rbush/issues/28
+        this.removeElement(spatialTree, elementId, startPosition);
+        this.addElement(spatialTree, elementId, targetPosition);
+    }
+
+    /**
+     * @param circlePosition the middle point of the search-circle
+     * @param radius of the search-circle
+     *
+     * @returns the ids of the elements in the search-circle, sorted by distance to {@link circlePosition}
+     */
+    public static findAllElementsInCircle(
+        spatialTree: Mutable<SpatialTree>,
+        circlePosition: Position,
+        radius: number
+    ): UUID[] {
+        if (radius <= 0) {
+            return [];
+        }
```

Comment:
Is this correct? Wouldn't `radius === 0` find elements that have exactly the same coordinates?

Reply:
See the other comment in `calculate-treatments.ts`.

Also `knn` handles `radius === 0` as `radius === infinity` resulting in not the correct result, thefore I had implemented before something like this:
https://github.com/hpi-sam/digital-fuesim-manv/commit/7c7254f33875a6ff542fbb727110ce0d1cfa5457

Maybe putting a comment here about what to consider

Reply:
I get you're reasoning, so please add a comment.